src/hippopt/turnkey_planners/humanoid_kinodynamic/main_periodic_step.py

```diff
@@ -0,0 +1,497 @@
+    settings.casadi_solver_options = {}
+
+    return settings
+
+
+def get_visualizer_settings(
+    input_settings: walking_settings.Settings,
+) -> hp_rp.HumanoidStateVisualizerSettings:
+    output_viz_settings = hp_rp.HumanoidStateVisualizerSettings()
+    output_viz_settings.robot_model = input_settings.robot_urdf
+    output_viz_settings.considered_joints = input_settings.joints_name_list
+    output_viz_settings.contact_points = input_settings.contact_points
+    output_viz_settings.terrain = input_settings.terrain
+    output_viz_settings.working_folder = "./"
+
+    return output_viz_settings
+
+
+def compute_state(
+    input_settings: walking_settings.Settings,
+    pf_input: pose_finder.Planner,
+    desired_com_position: np.ndarray,
+    desired_left_foot_pose: liecasadi.SE3,
+    desired_right_foot_pose: liecasadi.SE3,
+) -> hp_rp.HumanoidState:
+    desired_joints = np.deg2rad(
+        [
+            7,
+            0.12,
+            -0.01,
+            12.0,
+            7.0,
+            -12.0,
+            40.769,
+            12.0,
+            7.0,
+            -12.0,
+            40.769,
+            5.76,
+            1.61,
+            -0.31,
+            -31.64,
+            -20.52,
+            -1.52,
+            5.76,
+            1.61,
+            -0.31,
+            -31.64,
+            -20.52,
+            -1.52,
+        ]
+    )
+    assert len(input_settings.joints_name_list) == len(desired_joints)
+
+    pf_ref = pose_finder.References(
+        contact_point_descriptors=pf_settings.contact_points,
+        number_of_joints=len(desired_joints),
+    )
+
+    pf_ref.state.com = desired_com_position
+    pf_ref.state.contact_points.left = (
+        hp_rp.FootContactState.from_parent_frame_transform(
+            descriptor=input_settings.contact_points.left,
+            transform=desired_left_foot_pose,
+        )
+    )
+    pf_ref.state.contact_points.right = (
+        hp_rp.FootContactState.from_parent_frame_transform(
+            descriptor=input_settings.contact_points.right,
+            transform=desired_right_foot_pose,
+        )
+    )
+
+    pf_ref.state.kinematics.base.quaternion_xyzw = (
+        liecasadi.SO3.Identity().as_quat().coeffs()
+    )
+
+    pf_ref.frame_quaternion_xyzw = liecasadi.SO3.Identity().as_quat().coeffs()
+
+    pf_ref.state.kinematics.joints.positions = desired_joints
+
+    pf_input.set_references(pf_ref)
+
+    output_pf = pf_input.solve()
+    return output_pf.values.state
+
+
+def compute_initial_state(
+    input_settings: walking_settings.Settings,
+    pf_input: pose_finder.Planner,
+    contact_guess: hp_rp.FeetContactPhasesDescriptor,
+) -> walking_variables.ExtendedHumanoidState:
+    desired_left_foot_pose = contact_guess.left[0].transform
+    desired_right_foot_pose = contact_guess.right[0].transform
+    desired_com_position = (
+        desired_left_foot_pose.translation() + desired_right_foot_pose.translation()
+    ) / 2.0
```

Comment:
Just a minor comment. Here you have set the `desired_com_position` equal to the average of the origins of the two feet. Shouldn't the command `desired_com_position[2] = 0.7` be added as in the cases of `compute_middle_state()` and `compute_final_state()`?

Reply:
Nice catch, thanks! Fixed in https://github.com/ami-iit/hippopt/pull/12/commits/a37a6a2d5704d33818b2c9317342b989de1723c1